api/src/main/java/io/minio/CopyConditions.java

```diff
@@ -0,0 +1,131 @@
+/*Minio Java Library for Amazon S3 Compatible Cloud Storage,(C)2015 Minio,Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio;
+
+import java.util.Date;
+
+import io.minio.errors.InvalidArgumentException;
+
+/**
+ * A container class to hold all the Conditions to be checked before copying an object.
+ */
+public class CopyConditions {
+
+  Date dateUnmodifiedSince;
+  Date dateModifiedAfter;
+  String matchETag;
+  String exceptETag;
```

Comment:
Are these names from minio-go?  It would be clear name like (as similar as the spec)
```
String matchEtag;
String noneMatchEtag;
String unmodifiedSince;
String modifiedSince;
```

Also, please have setter methods names prefixing `set` and member variable name.